### PR TITLE
5/7/19 functional+cosmetic changes

### PR DESCRIPTION
**Pull request by Accio1**
@locness3 This is my most recent update. The change log is below. I figured you might want to review it. If you don't, that's fine as well.

Change Log:
javascript.js (Renamed later to main.js):
-Added a 404 page for if the user inputs an invalid Studio Id
-Added two shortcuts (SDS and Featured), which direct to the current SDS and Featured Studio

stylesheet.css:
-Added CSS for the error box
-Added min-height in order to make sure the box doesn't get too small

index.html:
-Added the place where an error message can be displayed.
-Added a default title of "Loading..."

(New File) home/home.js:
Moved JavaScript from script tags to a separate file

home/stylesheet.css:
-Added style for DevTag
-Removed blue glow from embed code textarea
-Changed <input type="submit"> to buttons to make things more consistent.

home/index.html:
-Added a version tag (Currently says STABLE)
-Made the current featured studio the default studio displayed
-Made text appear whenever someone copies something
-Move JavaScript to a separate file (home.js)
-Removed most CSS from <head> element.

### CODE REVIEW

**Review by moeenio — CHANGES_REQUESTED** [review]
The script doesn't load on home page, because it's pointing to the old javascript.js

**Comment by Accio1** [comment]
> The script doesn't load on home page, because it's pointing to the old javascript.js

Fixed

**Comment by Accio1** [comment]
Is everything else fine?

**Comment by moeenio** [comment]
I'm testing...

**Comment by moeenio** [comment]
The way you list changelog by file in your PR description is weird. Do a changelog per feature.

File: home/home.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+document.getElementById("Ecode").innerHTML = '<iframe src="https://accio1.github.io/SSE/#FEATURED" style="width: 250px; min-height: 538.5px; border: none;"></iframe>';
+	
+function GoTo() {
+	var GT_StudioID;
+	GT_StudioID = document.getElementById("GT_StudioID").value;
+		if (isNaN(GT_StudioID) && GT_StudioID !=="SDS" && GT_StudioID !=="FEATURED") {
+			document.getElementById("GT_result").innerHTML = "Not A Studio ID";
+  			} 
+		else {
+    			window.location.href = "/SSE/#"+GT_StudioID;
```
Comment (moeenio) [inline home/home.js]: Use `..` instead of `/SSE` (for testing locally or with Netlify)

File: home/index.html [hunk 2]
```diff
@@ -8,83 +8,35 @@
...
-			float: left;
-			background: #aae9ff;
-			}
 		</style>
 	</head>
 	<body style="margin:0">
 		<div class="content">
-			<h1>Scratch Studio Embed</h1>
+			<h1>Scratch Studio Embed
+			<span id="DevTag">STABLE</span></h1>
```
Comment (moeenio) [inline home/index.html]: Logically, the easier accessible version is the stable one. You should either remove that, or start using the GitHub 's releases feature and number versionning (following the model MAJOR.MINOR.PATCH), and indicate the release number here.

**Review by moeenio — CHANGES_REQUESTED** [review]

File: home/home.js (hunk 1 above)
Comment (Accio1) [reply home/home.js]: Done

File: home/index.html (hunk 2 above)
Comment (Accio1) [reply home/index.html]: Fixed, I made it display "By -Accio-" instead of "STABLE"

**Comment by Accio1** [comment]
Is it now ready to be merged?

File: main.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
+setTimeout(display, 50)
+
+if (window.location.hash === "" || window.location.hash === "#") {
+	window.location.replace("/SSE/home");
```
Comment (moeenio) [inline main.js]: Same here, don't add `/SSE`. in this case you can use `home` directly.

**Review by moeenio — CHANGES_REQUESTED** [review]

File: main.js (hunk 3 above)
Comment (Accio1) [reply main.js]: That doesn't work. It brings me to https://accio1.github.io/home

**Comment by Accio1** [comment]
Can it be merged?

**Comment by Accio1** [comment]
> Can it be merged?

If it's not ready, please suggest everything which needs to be changed all at once next time, as then I know I can merge as soon as I have implemented the suggestions.

File: main.js (hunk 3 above)
Comment (moeenio) [reply main.js]: Without a `/` at the beginning. or try `./home`

Comment (Accio1) [reply main.js]: I added a base element and made it just "home". I'm assuming this is the last thing that needed to be changed, as my comment below said:

"If it's not ready, please suggest everything which needs to be changed all at once next time, as then I know I can merge as soon as I have implemented the suggestions."

If there are no other suggestions within two hours, I will assume that was the last one and merge this.

Thanks for all of your help!

Comment (moeenio) [reply main.js]: Wait!

**Review by moeenio — CHANGES_REQUESTED** [review]
The idea with using relative paths is to **_never_** have an absolute URL. I just can't test locally, and you can't too with what we have (Did you even **try** testing it locally?). Everything is redirected to the current GitHub version. You should use relative paths **_everywhere_**. In relative paths, use :
- `/` for root
- `.` for current folder
- `..` for parent folder
- nothing for the current folder

**Comment by Accio1** [comment]
I did test it locally, by combining the stylesheet and the JavaScript into the index.html. Then I separated it when I put it on GitHub.

**Comment by moeenio** [comment]
Well, you shouldn't do that! You should use relative urls, so everyone can test it anywhere without playing around! I gave you a full guide in my review.

**Comment by Accio1** [comment]
That doesn’t work on Chromebooks. That’s why I do it the way I do.

**Comment by moeenio** [comment]
Deploy to Netlify then. All you have to do is to drag your site folder.

**Comment by Accio1** [comment]
I’d rather just test it locally on my Chromebook.

**Comment by moeenio** [comment]
Please, test it on Netlify. To make sure your relative urls and all work correctly, and to make sure it will work perfectly on GitHub pages.

**Comment by moeenio** [comment]
You can register on Netlify with your GitHub account, and once you have done so, you just have to **drag-and-drop the folder on Netlify's interface** to deploy your site.

**Comment by Accio1** [comment]
Yes, but I don’t mind just testing it on my Chromebook.

**Comment by moeenio** [comment]
That's bad practice. Sorry, but you really need to test it on a "real" environement. You can't ask people to review your changes if they can't do it the way they would do it. And you shouldn't rely on your GitHub URL for flexibility.

**Comment by Accio1** [comment]
You've convinced me. I made all of the URLs relative. Are you able to test it now?

**Comment by moeenio** [comment]
Yes! It works great

**Comment by moeenio** [comment]
As I already said, the way you do your changelog by file in your PR description is weird. Do a changelog per feature.

**Review by moeenio — APPROVED** [review]

**Comment by Accio1** [comment]
Change Log: 
-Added a 404 page if the studio id is invalid.
-Added shortcuts which can replace studio ids (Featured for the current featured studio, SDS for the current SDS).
-Added a min-height style property which makes sure the box is always a certain size.
-Made the default title "Loading..." so users know that it's loading.
-Moved the JavaScript for the homepage into a separate file instead of script tags.
-Changed `<input type="submit">` to buttons to make things more consistent.
-Added a little bubble next to the title which says "By: -Accio-"
-Added some text which appears when someone uses the copy button

Ready to merge now?

**Comment by Accio1** [comment]
I saw that you approved the changes! Merging!

**Comment by moeenio** [comment]
🎉

**Comment by moeenio** [comment]
BTW you could have edited your PR's description isntead of creating a new comment, but that's okay

**Pull request merged**